picamera2/request.py

```diff
@@ -235,17 +240,21 @@ def save(self, img, metadata, file_output, format=None):
                 # doesn't like RGBA to we have to bodge that to RGBX.
                 img.mode = "RGBX"
             # Make up some extra EXIF data.
-            zero_ifd = {piexif.ImageIFD.Make: "Raspberry Pi",
+            if "AnalogueGain" in metadata and "DigitalGain" in metadata:
+                zero_ifd = {piexif.ImageIFD.Make: "Raspberry Pi",
                         piexif.ImageIFD.Model: self.picam2.camera.id,
                         piexif.ImageIFD.Software: "Picamera2"}
-            total_gain = metadata["AnalogueGain"] * metadata["DigitalGain"]
-            exif_ifd = {piexif.ExifIFD.ExposureTime: (metadata["ExposureTime"], 1000000),
-                        piexif.ExifIFD.ISOSpeedRatings: int(total_gain * 100)}
-            exif = piexif.dump({"0th": zero_ifd, "Exif": exif_ifd})
+                total_gain = metadata["AnalogueGain"] * metadata["DigitalGain"]
```

Comment:
I think the line above (and the next few) are complaining about indentation and I suspect it's caused by lines 245 and 246 being insufficiently indented. Do you just want to double-check that? Thanks!